src/library/autodj/autodjprocessor.cpp

```diff
@@ -817,61 +965,204 @@ void AutoDJProcessor::calculateTransition(DeckAttributes* pFromDeck,
     double fromTrackDuration = pFromDeck->duration();
     double toTrackDuration = pToDeck->duration();
 
-    double fromTrackOutroStart = getOutroStartPosition(pFromDeck);
-    if (fromTrackOutroStart <= 0.0) {
-        fromTrackOutroStart = fromTrackDuration;
-    }
-
-    double fromTrackOutroEnd = getOutroEndPosition(pFromDeck);
-    if (fromTrackOutroEnd <= 0.0) {
-        fromTrackOutroEnd = fromTrackDuration;
+    VERIFY_OR_DEBUG_ASSERT(fromTrackDuration > 0) {
+        // Playing Track has no duration. This should not happen, because short
+        // tracks are skipped after load. Play ToDeck immediately.
+        pFromDeck->fadeBeginPos = 0;
+        pFromDeck->fadeEndPos = 0;
+        pToDeck->startPos = kKeepPosition;
+        return;
     }
-
-    double toTrackIntroStart = getIntroStartPosition(pToDeck);
-    if (toTrackIntroStart <= 0.0) {
-        toTrackIntroStart = 0.0;
+    VERIFY_OR_DEBUG_ASSERT(toTrackDuration > 0) {
+        // Playing Track has no duration. This should not happen, because short
+        // tracks are skipped after load.
+        pToDeck->startPos = kKeepPosition;
+        // Disable AutoDJ.
+        toggleAutoDJ(false);
     }
 
-    if (m_transitionTime > 0.0) {  // Crossfade transition
-        // Make sure length of selected outro is not negative.
-        double desiredTransitionTime = fromTrackOutroEnd - fromTrackOutroStart;
-        if (desiredTransitionTime <= 0.0) {
-            desiredTransitionTime = m_transitionTime;
+    // Within this function, the outro refers to the outro of the currently
+    // playing track and the intro refers to the intro of the next track.
+
+    double outroEnd = getOutroEndPosition(pFromDeck);
+    double outroStart = getOutroStartPosition(pFromDeck);
+    if (outroStart <= 0.0) {
+        // Assume a zero length outro.
+        // The outroEnd is automatically placed by AnalyzerSilence, so use
+        // that as a fallback if the user has not placed outroStart.
+        if (outroEnd >= 0) {
```

Comment:
This can be a debug assert, since it is handled in getOutroEndPosition()

Reply:
How about getting rid of the `if` statement and just using `outroStart = outroEnd;`?